src/enclave/enclave_oe.c

```diff
@@ -361,42 +388,112 @@ static void _copy_shared_memory(const sgxlkl_shared_memory_t* host)
         }
     }
 
-    if (host->env)
+    /* Import environment variables from the host */
+    const char* henv = host->env;
+    size_t henv_len = host->env_length;
+    if (cfg->num_host_import_env && cfg->host_import_env && henv && henv_len)
     {
-        size_t henvc = 0;
-        while (host->env[henvc++] != 0)
-            ;
+        sgxlkl_ensure_outside(henv, sizeof(char) * henv_len);
+
+        size_t num_vars = 0;
+        for (size_t i = 0; i < henv_len; i++)
+            if (henv[i] == '\0')
```

Comment:
What's the definition of `host->env`?

Reply:
I'm not sure I understand your question. `host->henv` is a long string of size `host->env_length`, which contains `\0` characters between environment variable definitions.

Reply:
Got it

Reply:
Sure, that's easy enough to add.